### PR TITLE
Add endpoint /devices/{ID}/details to get all data

### PR DESCRIPTION
**Pull request by matoval**
Co-authored-by: Demetrius <de1987@users.noreply.github.com>

# Description

Add endpoint /devices/{ID}/details to get all necessary data for the device table for the UI 

Fixes # ([THEEDGE-1831](https://issues.redhat.com/browse/THEEDGE-1831))

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I run `go fmt ./...` to check that my code is properly formatted
- [x] I run `go vet ./...` to check that my code is free of common Go style mistakes

### CODE REVIEW

File: pkg/services/devicegroups.go [hunk 1]
```diff
@@ -154,6 +158,40 @@ func (s *DeviceGroupsService) GetDeviceGroupByID(ID string) (*models.DeviceGroup
 	return &deviceGroup, nil
 }
 
+// GetDeviceGroupDetailsByID gets the device group details by ID from the database
+func (s *DeviceGroupsService) GetDeviceGroupDetailsByID(ID string) (*models.DeviceGroupDetails, error) {
+	var deviceGroupDetails models.DeviceGroupDetails
+	account, err := common.GetAccountFromContext(s.ctx)
+	if err != nil {
+		s.log.WithField("error", err.Error()).Error("Error account")
```
Comment (matoval) [inline pkg/services/devicegroups.go:166]: @acosferreira Is this the correct way to handle the error?

Comment (acosferreira) [reply pkg/services/devicegroups.go:166]: You should return the error as well to stop the process, we can't continue without an account.
i.e return nil, err

File: pkg/services/devicegroups.go [hunk 2]
```diff
@@ -154,6 +158,40 @@ func (s *DeviceGroupsService) GetDeviceGroupByID(ID string) (*models.DeviceGroup
...
+	}
+	result := db.DB.Where("account = ? and id = ?", account, ID).Preload("Devices").First(&deviceGroupDetails.DeviceGroup)
+	if result.Error != nil {
+		return nil, new(DeviceGroupNotFound)
```
Comment (acosferreira) [inline pkg/services/devicegroups.go:170]: Add the log here as well

File: pkg/services/devicegroups.go [hunk 3]
```diff
@@ -154,6 +158,40 @@ func (s *DeviceGroupsService) GetDeviceGroupByID(ID string) (*models.DeviceGroup
...
+	}
+
+	if len(deviceGroupDetails.DeviceGroup.Devices) > 0 {
+		var devices models.DeviceDetailsList
+		for _, device := range deviceGroupDetails.DeviceGroup.Devices {
+			param := new(inventory.Params)
+			param.HostnameOrID = device.UUID
+			inventoryDevice, err := s.DeviceService.GetDevices(param)
+			if err != nil {
+				s.log.WithField("error", err.Error()).Error("Invetory error")
```
Comment (acosferreira) [inline pkg/services/devicegroups.go:180]: If an error occurs, we want to continue or stop and respond the error?

**Review by acosferreira — APPROVED** [review]
lgtm

**Pull request merged**